cuda_core/docs/design/buffer-fill-redesign.md

```diff
@@ -0,0 +1,350 @@
+## Design Options
+
+### Option A: Dtype-Inferred Fill (Leo's Approach)
+
+**`Buffer.fill()`**:
+```python
+def fill(self, value, *, stream: Stream | GraphBuilder):
+    """Fill buffer with value, inferring width from value's type.
+    
+    Parameters
+    ----------
+    value : scalar (int, numpy scalar, or bytes)
+        - Python int: treated as 1-byte unsigned (0-255)
+        - numpy.int8/uint8: 1-byte fill
+        - numpy.int16/uint16: 2-byte fill  
+        - numpy.int32/uint32: 4-byte fill
+        - bytes (length 1, 2, or 4): raw byte pattern
+    stream : Stream | GraphBuilder
+        Stream for asynchronous operation
+    """
+```
+
+**Pros**:
+- Uniform UX with `cuda.core.launch()` scalar handling
+- No explicit width parameter needed
+- Signedness is explicit via dtype
+
+**Cons**:
+- Plain Python `int` behavior needs definition (default to what width?)
+- Requires NumPy dependency awareness
+
+### Option B: Split API (Keith's Approach)
+
+**`Buffer.fill()`** - Byte-only:
+```python
+def fill(self, value: int | bytes, *, stream: Stream | GraphBuilder):
+    """Fill buffer with single-byte value.
+    
+    Parameters
+    ----------
+    value : int (0-255) or bytes (length 1)
+        Byte value to fill with
+    """
+```
+
+**`StridedMemoryView.fill()`** - Dtype-aware:
+```python
+def fill(self, value, *, stream: Stream | GraphBuilder):
+    """Fill view with value using view's dtype for width/signedness."""
+```
+
+**Pros**:
+- Clear separation of concerns
+- Buffer stays simple (untyped bytes)
+- StridedMemoryView has dtype context naturally
+
+**Cons**:
+- Two different APIs for similar operations
+- Users must create StridedMemoryView for multi-byte fills
+
+### Option C: Hybrid Approach
+
+**`Buffer.fill()`** - Flexible:
+```python
+def fill(self, value, *, stream: Stream | GraphBuilder):
+    """Fill buffer with value pattern.
+    
+    Parameters
+    ----------
+    value : int, numpy scalar, or bytes
+        - int: 1-byte fill (0-255)
+        - numpy.{u}int{8,16,32}: corresponding width fill
+        - bytes (1-4 bytes): raw byte pattern fill
+    """
+```
+
+**`StridedMemoryView.fill()`** - Uses dtype:
+```python
+def fill(self, value, *, stream: Stream | GraphBuilder):
+    """Fill view elements with value (uses view's dtype)."""
+```
+
+**Pros**:
+- Buffer API is flexible and discoverable
+- StridedMemoryView uses its natural dtype context
+- Consistent with launch() scalar handling
+
+**Cons**:
+- More complex implementation in Buffer
+
+## Recommendation
+
+**Option C (Hybrid Approach)** provides the best balance:
+
+1. **Buffer.fill()** accepts:
+   - `int` → 1-byte fill (value 0-255)
+   - `numpy.int8/uint8` → 1-byte fill
+   - `numpy.int16/uint16` → 2-byte fill
+   - `numpy.int32/uint32` → 4-byte fill
+   - `bytes` (length 1, 2, or 4) → raw pattern fill
+
+2. **StridedMemoryView.fill()** (new):
+   - Uses the view's dtype to determine width and signedness
+   - Value is validated against the dtype's range
```

Comment:
If the `StridedMemoryView` is non-contiguous, do we not support it for now or does the fill operate ignoring strides?